plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/generator/SQLGeneratorColumnNameTypeNN.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.sql.generator;
+
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSEntityAttribute;
+
+public class SQLGeneratorColumnNameTypeNN extends SQLGeneratorColumnNameType {
```

Comment:
And it would be better if this class was called `SQLGeneratorColumnNameTypeNotNull` instead.

Now I have a question, do we really need a separate generator for `NOT NULL`? Can't we do that always when the attribute is required?

Reply:
The question is hard enough, but let us face it up. When we need "Name + Type" or something else? Simply - it is useful. When we want to create scripts more quickly. Sometimes we need strict rules, as "not null", other times - not, when we need to load temp data, that looks like existing table, but we not sure about nulls and other checks. 
Dont forget about "default" - i missed it, but not forget )))
What is the purpose of "copy" or "copy full data" in the fieldlist? both are not so convient.

Should I remove one of these?

Reply:
I dunno.

It seems like in the future we might end implementing another generator that will have the `DEFAULT` modifier. And they will multiply endlessly because we will want to have `DEFAULT` with `NOT NULL`, or without, etc.

I think the best solution would be just to include `NOT NULL` by default (if the attribute is indeed "required").

Reply:
So we can call a table ddl call - then copy columns, that we need, then remove null / not null modifiers when we need / dont need them, remove default modifier in similar circumstances, remove commas if we need / or not to insert data into table.
We can do all of above.
There is 0 (zero) generators exists for DBSEntityAttribute. 
Do the growth for a two is too much for it for a 2.5 years of initial ticket life? 
Is there a chance for them to "multiply endlessly" ? 

I dunno )))))

Reply:
Okay, then it seems fine.